src/scheduler/node_info.cpp

```diff
@@ -1071,6 +1071,11 @@ remove_node_state(node_info *ninfo, const char *state)
 		&& !ninfo->is_unknown  && !ninfo->is_down && !ninfo->is_maintenance)
 		ninfo->is_free = 1;
 
+	if (ninfo->is_free)
+		ninfo->nscr &= ~NSCR_INELIGIBLE;
+	else
+		ninfo->nscr |= NSCR_INELIGIBLE;
+
```

Comment:
There are quite a few more reasons a node can be marked ineligible than just state.  If you want to do this, you need to add a new element to the enum and create a ne NSCR bit.